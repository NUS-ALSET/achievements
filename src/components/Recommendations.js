/**
 * @file Recommendations container module
 * @author Theodor Shaytanov <theodor.shaytanov@gmail.com>
 * @created 13.06.18
 */

import React, { Fragment } from "react";
import PropTypes from "prop-types";

import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";

import PathCard from "./cards/PathCard";

class Recommendations extends React.PureComponent {
  static propTypes = {
    recs: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired
  };

  getRecTitle = rec => `Use the ${rec.feature} ${rec.featureType}`;

  render() {
    const { recs, title } = this.props;

    return (
      <Fragment>
        <Typography
          style={{
            marginLeft: 4
          }}
          variant="headline"
        >
          {title}
        </Typography>
        <Paper
          style={{
            marginBottom: 24
          }}
        >
          <Grid container spacing={4}>
            <Grid item sm={4} xs={12}>
              <PathCard
                featured={true}
                path={recs[0].path}
                problem={recs[0].problem}
                title={this.getRecTitle(recs[0])}
                video={recs[0].video}
              />
            </Grid>
            <Grid
              item
              sm={8}
              style={{
                display: "flex",
                flexWrap: "wrap"
              }}
              xs={12}
            >
              {recs.slice(1).map(rec => (
                <PathCard
                  key={rec.problem}
                  path={rec.path}
                  problem={rec.problem}
                  title={this.getRecTitle(rec)}
                  video={rec.video}
                />
              ))}
            </Grid>
          </Grid>
        </Paper>
      </Fragment>
    );
  }
}

export default Recommendations;
