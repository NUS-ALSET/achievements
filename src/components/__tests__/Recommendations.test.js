import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import Recommendations from "../Recommendations";

describe("<Recommendations/>", () => {
  let shallow;
  beforeEach(() => {
    shallow = createShallow();
  });

  it("Should test Recommendations component", () => {
    const rec=[{
      problem : null,
      path : null,
      video : null,
      feature : null,
      featureType : null
    }];
    const component = shallow(
      <Recommendations  recs={rec}  title={""} />
    );
    expect(component).toMatchSnapshot();
  });

});
