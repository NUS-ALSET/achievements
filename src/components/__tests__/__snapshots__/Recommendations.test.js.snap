// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recommendations/> Should test Recommendations component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Recommendations
    recs={
        Array [
            Object {
              "feature": null,
              "featureType": null,
              "path": null,
              "problem": null,
              "video": null,
            },
          ]
    }
    title=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          style={
                    Object {
                              "marginLeft": 4,
                            }
          }
          variant="headline"
/>,
        <WithStyles(Paper)
          style={
                    Object {
                              "marginBottom": 24,
                            }
          }
>
          <GridWrapper
                    container={true}
                    spacing={4}
          >
                    <GridWrapper
                              item={true}
                              sm={4}
                              xs={12}
                    >
                              <PathCard
                                        featured={true}
                                        title="Use the null null"
                                        video={null}
                              />
                    </GridWrapper>
                    <GridWrapper
                              item={true}
                              sm={8}
                              style={
                                        Object {
                                                  "display": "flex",
                                                  "flexWrap": "wrap",
                                                }
                              }
                              xs={12}
                    >
                              
                    </GridWrapper>
          </GridWrapper>
</WithStyles(Paper)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "style": Object {
            "marginLeft": 4,
          },
          "variant": "headline",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <GridWrapper
            container={true}
            spacing={4}
>
            <GridWrapper
                        item={true}
                        sm={4}
                        xs={12}
            >
                        <PathCard
                                    featured={true}
                                    title="Use the null null"
                                    video={null}
                        />
            </GridWrapper>
            <GridWrapper
                        item={true}
                        sm={8}
                        style={
                                    Object {
                                                "display": "flex",
                                                "flexWrap": "wrap",
                                              }
                        }
                        xs={12}
            >
                        
            </GridWrapper>
</GridWrapper>,
          "style": Object {
            "marginBottom": 24,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridWrapper
                item={true}
                sm={4}
                xs={12}
>
                <PathCard
                                featured={true}
                                title="Use the null null"
                                video={null}
                />
</GridWrapper>,
              <GridWrapper
                item={true}
                sm={8}
                style={
                                Object {
                                                "display": "flex",
                                                "flexWrap": "wrap",
                                              }
                }
                xs={12}
>
                
</GridWrapper>,
            ],
            "container": true,
            "spacing": 4,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <PathCard
                  featured={true}
                  title="Use the null null"
                  video={null}
/>,
                "item": true,
                "sm": 4,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "featured": true,
                  "title": "Use the null null",
                  "video": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [],
                "item": true,
                "sm": 8,
                "style": Object {
                  "display": "flex",
                  "flexWrap": "wrap",
                },
                "xs": 12,
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            style={
                        Object {
                                    "marginLeft": 4,
                                  }
            }
            variant="headline"
/>,
          <WithStyles(Paper)
            style={
                        Object {
                                    "marginBottom": 24,
                                  }
            }
>
            <GridWrapper
                        container={true}
                        spacing={4}
            >
                        <GridWrapper
                                    item={true}
                                    sm={4}
                                    xs={12}
                        >
                                    <PathCard
                                                featured={true}
                                                title="Use the null null"
                                                video={null}
                                    />
                        </GridWrapper>
                        <GridWrapper
                                    item={true}
                                    sm={8}
                                    style={
                                                Object {
                                                            "display": "flex",
                                                            "flexWrap": "wrap",
                                                          }
                                    }
                                    xs={12}
                        >
                                    
                        </GridWrapper>
            </GridWrapper>
</WithStyles(Paper)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "style": Object {
              "marginLeft": 4,
            },
            "variant": "headline",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <GridWrapper
              container={true}
              spacing={4}
>
              <GridWrapper
                            item={true}
                            sm={4}
                            xs={12}
              >
                            <PathCard
                                          featured={true}
                                          title="Use the null null"
                                          video={null}
                            />
              </GridWrapper>
              <GridWrapper
                            item={true}
                            sm={8}
                            style={
                                          Object {
                                                        "display": "flex",
                                                        "flexWrap": "wrap",
                                                      }
                            }
                            xs={12}
              >
                            
              </GridWrapper>
</GridWrapper>,
            "style": Object {
              "marginBottom": 24,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <GridWrapper
                  item={true}
                  sm={4}
                  xs={12}
>
                  <PathCard
                                    featured={true}
                                    title="Use the null null"
                                    video={null}
                  />
</GridWrapper>,
                <GridWrapper
                  item={true}
                  sm={8}
                  style={
                                    Object {
                                                      "display": "flex",
                                                      "flexWrap": "wrap",
                                                    }
                  }
                  xs={12}
>
                  
</GridWrapper>,
              ],
              "container": true,
              "spacing": 4,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <PathCard
                    featured={true}
                    title="Use the null null"
                    video={null}
/>,
                  "item": true,
                  "sm": 4,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "featured": true,
                    "title": "Use the null null",
                    "video": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [],
                  "item": true,
                  "sm": 8,
                  "style": Object {
                    "display": "flex",
                    "flexWrap": "wrap",
                  },
                  "xs": 12,
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
